File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlAssignExpr.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlAssignExpr implements ISqlExpression {
    SqlType: SqlType = SqlType.AssignExpr;
    Span: TextSpan = new TextSpan();
    Left!: ISqlExpression;
    Right!: ISqlExpression;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_AssignExpr(this);
    }

    ToSql(): string {
        return `${this.Left.ToSql()} = ${this.Right.ToSql()}`;
    }
} 

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlAssignExpr.cs
namespace T1.SqlSharp.Expressions;

public class SqlAssignExpr : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.AssignExpr;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_AssignExpr(this);
    }

    public required ISqlExpression Left { get; set; }
    public required ISqlExpression Right { get; set; }

    public string ToSql()
    {
        return $"{Left.ToSql()} = {Right.ToSql()}";
    }
}